File: bro/src/main/java/me/xx2bab/bro/core/api/ApiRudder.java
package me.xx2bab.bro.core.api;

import me.xx2bab.bro.common.IBroApi;
import me.xx2bab.bro.core.BroContext;

public class ApiRudder {

    private IApiFinder annoApiFinder;

    public ApiRudder(BroContext broContext) {
        // todo: multiple impl
        annoApiFinder = new AnnoApiFinder(broContext);
    }

    public <T extends IBroApi> T getApi(Class<T> apiInterface) {
        return annoApiFinder.getApi(apiInterface);
    }

    public IBroApi getApi(String nick) {
        return annoApiFinder.getApi(nick);
    }


}

File: bro/src/main/java/me/xx2bab/bro/core/api/ApiRudder.kt
package me.xx2bab.bro.core.api

import me.xx2bab.bro.common.IBroApi
import me.xx2bab.bro.core.BroContext

class ApiRudder(broContext: BroContext) {
    private val annoApiFinder: IApiFinder

    init { // todo: multiple impl
        annoApiFinder = AnnoApiFinder(broContext)
    }

    fun <T : IBroApi> getApi(apiInterface: Class<T>): T? {
        return annoApiFinder.getApi(apiInterface)
    }

    fun getApi(nick: String): IBroApi? {
        return annoApiFinder.getApi(nick)
    }

}